lib/yocaml/metadata.ml

```diff
@@ -127,28 +130,22 @@ module Article = struct
     }
   ;;
 
-  let from_yaml yaml =
-    let open Validate in
-    Yaml.as_object
-      yaml
-      (fun obj ->
-        let open Applicative in
-        make
-        <$> Yaml.(required string "article_title" obj)
-        <*> Yaml.(required string "article_description" obj)
-        <*> Yaml.(with_default ~default:[] (list string) "tags" obj)
-        <*> Yaml.(required Date.from_yaml "date" obj)
-        <*> Yaml.(optional string "title" obj)
-        <*> Yaml.(optional string "description" obj))
-      (Validate.error $ Error.Invalid_metadata "Article")
-  ;;
-
-  let from_string = function
+  let from_string (module V : PROVIDER) = function
     | None -> Validate.error $ Error.Invalid_metadata "Article"
```

Comment:
I think that "Invalid_metadata" is not a good name for an error (but that's my fault). It should be an error that indicates that the document must be attached to metadata.

Reply:
The error `Required_metadata []` already exists so I use it here.

Reply:
Nice